src/promptflow-tools/promptflow/tools/common.py

```diff
@@ -442,6 +444,55 @@ def render_jinja_template(prompt, trim_blocks=True, keep_trailing_newline=True,
         raise JinjaTemplateError(message=error_message) from e
 
 
+def render_jinja_template_wrapper(prompt, trim_blocks=True, keep_trailing_newline=True, **kwargs):
+    # Use this dict to store the user written roles in case sensitive and its escaped value.
+    # We need this dict to unescape the role after the chat string is parsed.
+    escape_dict = {}
+    if isinstance(kwargs, dict):
+        updated_kwargs = {key: escape_roles(value, escape_dict) for key, value in kwargs.items()}
+    return (
+        render_jinja_template(
+            prompt,
+            trim_blocks=trim_blocks,
+            keep_trailing_newline=keep_trailing_newline,
+            **updated_kwargs,
+        ),
+        escape_dict,
+    )
+
+
+def escape_roles(val, escape_dict: dict):
```

Comment:
The way to implicitly modifying an input variable looks dangerous, and may causing potential breaking if people do not maintain well. Please help think of a better way.

Reply:
seperate escape_role and build_escape_dict to two functions